File: VB/ClientServer_Wcf_StandardAuth.Module/Controllers/ExceptionDemonstrateController.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports ClientServer_Wcf_StandardAuth.Module.BusinessObjects

Namespace ClientServer_Wcf_StandardAuth.Module.Controllers
    Public Class ExceptionDemonstrateController
        Inherits ViewController

        Public Sub New()
            TargetObjectType = GetType(TestBO)
            Dim action As New SimpleAction(Me, "Modify Name programmatically", DevExpress.Persistent.Base.PredefinedCategory.RecordEdit)
            action.SelectionDependencyType = SelectionDependencyType.RequireMultipleObjects
            AddHandler action.Execute, AddressOf action_Execute
        End Sub

        Private Sub action_Execute(ByVal sender As Object, ByVal e As SimpleActionExecuteEventArgs)
            Try
                CType(View.CurrentObject, TestBO).Name = "Modify"
                View.ObjectSpace.CommitChanges()
            Catch e1 As Exception
                View.ObjectSpace.Rollback()
                Throw
            End Try

        End Sub
    End Class
End Namespace

File: CS/ClientServer_Wcf_StandardAuth.Module/Controllers/ExceptionDemonstrateController.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using ClientServer_Wcf_StandardAuth.Module.BusinessObjects;

namespace ClientServer_Wcf_StandardAuth.Module.Controllers {
    public class ExceptionDemonstrateController : ViewController {
        public ExceptionDemonstrateController() {
            TargetObjectType = typeof(TestBO);
            SimpleAction action = new SimpleAction(this, "Modify Name programmatically", DevExpress.Persistent.Base.PredefinedCategory.RecordEdit);
            action.SelectionDependencyType = SelectionDependencyType.RequireMultipleObjects;
            action.Execute += new SimpleActionExecuteEventHandler(action_Execute);
        }

        void action_Execute(object sender, SimpleActionExecuteEventArgs e) {
            try {
                ((TestBO)View.CurrentObject).Name = "Modify";
                View.ObjectSpace.CommitChanges();
            } catch (Exception) {
                View.ObjectSpace.Rollback();
                throw;
            }

        }
    }
}
